### PR TITLE
Download JVM dependencies via Gradle

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

Gradle is a modern build tool for JVM languages: https://gradle.org/

Because Downloading dependencies via Gradle is faster than Maven,
this change replaces all `images/**/pom.xml` files with `build.gradle`.
(Dependabot supports also Gradle)

> Link related issues or pull requests, e.g. `Fix #123`, `Related to #456`, or `None`.

Related to #1000

> Check the following items (please remove needless items).

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

File: images/Dockerfile.java.erb [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+ARG GRADLE_BINARY_URL=https://services.gradle.org/distributions/${GRADLE_BINARY_FILE}
+RUN cd "${RUNNER_USER_HOME}" && \
+    curl -fsSLO --compressed "${GRADLE_BINARY_URL}" && \
+    curl -fsSL  --compressed "${GRADLE_BINARY_URL}.sha256" | \
+      xargs -I {} echo "{} *${GRADLE_BINARY_FILE}" | \
+      sha256sum --check --strict && \
+    unzip -q "${GRADLE_BINARY_FILE}" && \
+    mv "gradle-${GRADLE_VERSION}" gradle && \
+    rm "${GRADLE_BINARY_FILE}" && \
+    gradle -v
```
Comment (ybiquitous) [inline images/Dockerfile.java.erb:16]: [note] If this PR has no problems and will be merged, I will move this installation process to the sider/devon_rex repository.

File: images/Dockerfile.java.erb [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+# Install dependencies via Gradle
+ARG DEPS_DIR=${RUNNER_USER_HOME}/dependencies
+ENV CLASSPATH ${DEPS_DIR}/*:${CLASSPATH}
+COPY --chown=<%= chown %> images/<%= analyzer %>/build.gradle ${DEPS_DIR}/
+RUN cd "${DEPS_DIR}" && \
+    echo 'task deps(type: Copy) {' >> build.gradle && \
+    echo '  from configurations.runtimeClasspath' >> build.gradle && \
+    echo '  into "."' >> build.gradle && \
+    echo '}' >> build.gradle && \
+    gradle --no-build-cache --parallel --quiet deps && \
```
Comment (ybiquitous) [inline images/Dockerfile.java.erb:27]: [note] Create and run a custom task in Dockerfile.

**Comment by ybiquitous** [comment]
Download time is a bit faster on CI. For example, detekt is faster in a few seconds.

Before: **14.1s**
https://github.com/sider/runners/runs/923238658?check_suite_focus=true#step:7:185

After: **11.3s**
https://github.com/sider/runners/pull/1352/checks?check_run_id=923892918#step:7:166

But, on my local machine, the time diff is about **60s**.

File: images/Dockerfile.java.erb (hunk 1 above)
Comment (yykamei) [reply images/Dockerfile.java.erb:18]: I agree. Migrating the relevant code to sider/devon_rex seems good.

**Comment by ybiquitous** [comment]
Thank you for the review. I will move the code to sider/devon_rex later. 👍

**Pull request merged**